src/inputs/MultiLineSelectField.tsx

```diff
@@ -0,0 +1,118 @@
+import { useState } from "react";
+import { Button } from "src/components/Button";
+import { Label } from "src/components/Label";
+import { SelectField, Value } from "src/inputs";
+import { BeamSelectFieldBaseProps } from "src/inputs/internal/SelectFieldBase";
+import { Optional } from "src/types";
+import { Css, useTestIds } from "..";
+
+export interface MultiLineSelectFieldProps<O, V extends Value>
+  extends Exclude<BeamSelectFieldBaseProps<O, V>, "unsetLabel"> {
+  values: V[];
+  options: O[];
+  getOptionValue: (opt: O) => V;
+  getOptionLabel: (opt: O) => string;
+  onSelect: (values: V[], opts: O[]) => void;
+}
+
+export function MultiLineSelectField<O, V extends Value>(
+  props: Optional<MultiLineSelectFieldProps<O, V>, "getOptionLabel" | "getOptionValue">,
+): JSX.Element {
+  const {
+    options,
+    values,
+    onSelect,
+    getOptionValue = (opt: O) => (opt as any).id,
+    getOptionLabel = (opt: O) => (opt as any).name,
+    labelStyle,
+    ...otherProps
+  } = props;
+
+  const tid = useTestIds(props, "");
+  const [isDisplayed, setIsDisplayed] = useState(true);
+  // Set the available options for the current input field, initialize by filtering already selected options
+  const [currentOptions, setCurrentOptions] = useState(options.filter((o) => !values.includes(getOptionValue(o))));
+
+  return (
+    <div css={Css.mt1.if(labelStyle === "left").df.$}>
+      {labelStyle !== "hidden" && (
+        <div css={Css.if(labelStyle === "left").w50.$}>
+          <Label {...tid.label} label={props.label} />
+        </div>
+      )}
+      <div css={Css.if(labelStyle === "left").w50.$}>
+        {values.map((value, index) => {
+          return (
+            <div css={Css.mb1.pl1.df.$} key={index}>
+              <div css={Css.truncate.w100.$}>
+                <SelectField
+                  {...otherProps}
+                  {...tid.selectField}
+                  labelStyle="hidden"
+                  value={value}
+                  onSelect={() => {}}
+                  options={options}
+                  getOptionValue={getOptionValue}
+                  getOptionLabel={getOptionLabel}
+                  compact={true}
+                  readOnly={true}
+                />
+              </div>
+              <Button
+                {...tid.deleteSelected}
+                variant="tertiary"
+                label={""}
+                aria-label={`Delete selected ${otherProps.label}`}
+                icon={"x"}
+                onClick={() => {
+                  // Delete the selected value from the array
+                  const [selectedValues, selectedOptions] = options
+                    .filter((o) => values.filter((v) => v !== value).includes(getOptionValue(o)))
```

Comment:
Suggestion: We're invoking `values.filter(v => v !== value)` inside of a `filter(o => ...)` but not using the `o`, which means we're re-doing the same `values.filter` over and over w/o needing to.

In general it seems like this could be simpler, like something like:

```
onClick={() => {
  const selectedOptions = options.filter(o => getOptionValue(o) !== value);
  const selectedValues = selectedOptions.map(o => getOptionValue(o));
  onSelect(selectedValues, selectedOptions);
```

Granted, we're invoking `getOptionValue` twice for each `o`, but I think the current code is also doing that, and :shrug: `getOptionValue` should be really cheap to invoke.

Reply:
In this case  `const selectedOptions = options.filter(o => getOptionValue(o) !== value)` would make it select _all_ options instead of the current deleted, I updated it to take into account only selected values
```
const selectedOptions = options.filter( 
   (o) => values.includes(getOptionValue(o)) && getOptionValue(o) !== value,
);
```

Reply:
@roberthdg ah right! Good catch!